decidim-proposals/app/controllers/decidim/proposals/admin/proposal_answers_controller.rb

```diff
@@ -44,6 +63,42 @@ def skip_manage_component_permission
         def proposal
           @proposal ||= Proposal.where(component: current_component).find(params[:id])
         end
+
+        def proposals
+          @proposals ||= Proposal.where(component: current_component).where(id: params[:proposal_ids])
+        end
+
+        def template
+          @template ||= Decidim::Templates::Template.find(params[:template][:template_id])
+        end
+
+        def bulk_answer_form(proposal)
+          @bulk_answer_form ||= form(ProposalAnswerForm).from_params(prepare_answer_form_params(template, proposal, current_user))
+        end
+
+        def prepare_answer_form_params(template, proposal, current_user)
+          answer_form_params = {
+            answer: translated_attribute(template.description),
+            internal_state: Decidim::Proposals::ProposalState.find(template.field_values["proposal_state_id"]).token,
+            current_user:
+          }
+
+          if current_component.current_settings.answers_with_costs?
+            [:cost, :cost_report, :execution_period].each do |field|
+              value = proposal.send(field)
+              answer_form_params[field] = translated_attribute(value) if value.present?
+            end
+          end
+
+          answer_form_params
+        end
+
+        def missing_cost_data?(proposals)
+          proposals.each do |proposal|
+            return true if bulk_answer_form(proposal).costs_required? && proposal.cost.blank?
+          end
+          false
+        end
```

Comment:
```suggestion
        def answer_forms_invalid?(proposals)
          proposals.each do |proposal|
            return true if bulk_answer_form(proposal).invalid?
          end
          false
        end
```
This should be enough and it's more generic no?